lib/grammar.pegjs

```diff
@@ -810,6 +810,12 @@ time_value = ('makeTime' / 'new' __ 'Time') _ '(' hour:literal_number _ ',' _ mi
 } / '$context' _ '.' _ 'time' _ '.' _ ctx:('morning' / 'evening') {
     return new Ast.Value.Time(new Ast.Time.Relative(ctx));
 }
+piece_date_value = ('makeDatePiece' / 'new' __ 'DatePiece') _ '(' _ year:(literal_number _)? ',' _ month:(literal_number _)? ',' _ day:(literal_number _)? ',' _ time:(time_value _)? ')' {
```

Comment:
`makeX` is our old syntax, it's deprecated everywhere, we don't need it in new stuff.

Also, if you put `_` outside of the parenthesis and write `year:literal_number? _` you can simplify how you check that the year is present or not, because a missing value will be null, and `_` (whitespace) is always optional.